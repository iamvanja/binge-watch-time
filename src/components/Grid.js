import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

const GridContainer = ({ children, className, fluid, full, ...props }) => {
  const classes = classnames('grid-container', { fluid, full }, className)

  return (
    <div {...props} className={classes}>
      {children}
    </div>
  )
}

GridContainer.propTypes = {
  fluid: PropTypes.bool,
  full: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.node
}

const getDirectionClass = (isVertical, gutters = null) => {
  return isVertical === true
    ? gutters
      ? `grid-${gutters}-y`
      : 'grid-y'
    : gutters
      ? `grid-${gutters}-x`
      : 'grid-x'
}

const horizontalAlignments = [
  'center',
  'right',
  'justify',
  'spaced'
]

const verticalAlignments = [
  'top',
  'middle',
  'bottom',
  'stretch'
]

const Grid = ({ children, vertical, gutters, align, className, ...props }) => {
  const classes = classnames(
    getDirectionClass(vertical),
    gutters && getDirectionClass(vertical, gutters),
    align && `align-${align}`,
    className
  )

  return (
    <div {...props} className={classes}>
      {children}
    </div>
  )
}

Grid.propTypes = {
  vertical: PropTypes.bool,
  gutters: PropTypes.oneOf(['margin', 'padding']),
  align: PropTypes.oneOf([...horizontalAlignments, ...verticalAlignments]),
  children: PropTypes.node,
  className: PropTypes.string
}

const Cell = ({
  children,
  className,
  small,
  medium,
  large,
  alignSelf,
  ...props
}) => {
  const classes = classnames(
    'cell',
    small && `small-${small}`,
    medium && `medium-${medium}`,
    large && `large-${large}`,
    alignSelf && `align-self-${alignSelf}`,
    className
  )

  return (
    <div {...props} className={classes}>
      {children}
    </div>
  )
}

Cell.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  small: PropTypes.number,
  medium: PropTypes.number,
  large: PropTypes.number,
  alignSelf: PropTypes.oneOf([...horizontalAlignments, ...verticalAlignments])
}

export { GridContainer, Grid, Cell }
