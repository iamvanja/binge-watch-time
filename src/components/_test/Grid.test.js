import React from 'react'
import { shallow } from 'enzyme'
import { GridContainer, Grid, Cell } from '../Grid'

describe('GridContainer', () => {
  it('always renders a GridContainer', () => {
    const wrapper = shallow(
      <GridContainer>children</GridContainer>
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('always renders a GridContainer with props', () => {
    const wrapper = shallow(
      <GridContainer fluid full>children</GridContainer>
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Grid', () => {
  it('always renders a Grid', () => {
    const wrapper = shallow(
      <Grid vertical gutters='padding' align='middle'>
        <Cell>cell</Cell>
      </Grid>
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('Cell', () => {
  it('always renders a Cell', () => {
    const wrapper = shallow(
      <Cell small={12} medium={6} alignSelf='top'>cell</Cell>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
