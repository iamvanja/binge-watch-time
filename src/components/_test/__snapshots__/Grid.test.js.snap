// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell always renders a Cell 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cell
    alignSelf="top"
    medium={6}
    small={12}
>
    cell
</Cell>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "cell",
      "className": "cell small-12 medium-6 align-self-top",
    },
    "ref": null,
    "rendered": "cell",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "cell",
        "className": "cell small-12 medium-6 align-self-top",
      },
      "ref": null,
      "rendered": "cell",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Grid always renders a Grid 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid
    align="middle"
    gutters="padding"
    vertical={true}
>
    <Cell>
        cell
    </Cell>
</Grid>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Cell>
        cell
</Cell>,
      "className": "grid-y grid-padding-y align-middle",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "cell",
      },
      "ref": null,
      "rendered": "cell",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Cell>
          cell
</Cell>,
        "className": "grid-y grid-padding-y align-middle",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "cell",
        },
        "ref": null,
        "rendered": "cell",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GridContainer always renders a GridContainer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridContainer>
    children
</GridContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "children",
      "className": "grid-container",
    },
    "ref": null,
    "rendered": "children",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "children",
        "className": "grid-container",
      },
      "ref": null,
      "rendered": "children",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`GridContainer always renders a GridContainer with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GridContainer
    fluid={true}
    full={true}
>
    children
</GridContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "children",
      "className": "grid-container fluid full",
    },
    "ref": null,
    "rendered": "children",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "children",
        "className": "grid-container fluid full",
      },
      "ref": null,
      "rendered": "children",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
